src/android/CodePushReportingManager.java

```diff
@@ -13,28 +12,7 @@
 
     private Activity cordovaActivity;
     private CodePushPreferences codePushPreferences;
-
-    /**
-     * Defines application statuses we use in reporting events from the native to the JS layer.
-     */
-    public static enum Status {
-        STORE_VERSION(0),
-        UPDATE_CONFIRMED(1),
-        UPDATE_ROLLED_BACK(2);
-
-        private int value;
-
-        Status(int i) {
-            this.value = i;
-        }
-
-        /**
-         * Returns the value associated with this enum.
-         */
-        public int getValue() {
-            return this.value;
-        }
-    }
+    private Boolean hasFailedReport = null;
```

Comment:
Checking prefs means reading data from disk, which could be a fairly long operation and I prefer not to do that everytime the app resumes. We don't want to be responsible for slow performance on apps using our plugin.